From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Janet Blackquill <uhhadd@gmail.com>
Date: Sun, 7 Apr 2024 16:51:52 -0400
Subject: [PATCH] Add get/setResult for CartographyInventory

These are nice to have, and also necessary for Paper-Server
to tell if it should fire a CartographyItemEvent.

diff --git a/src/main/java/org/bukkit/inventory/CartographyInventory.java b/src/main/java/org/bukkit/inventory/CartographyInventory.java
index 29c9b2682b92433f468d434d25d3c2495b5ac91b..f333fc8adf6f57dd7d431ebcbb4736ac59962784 100644
--- a/src/main/java/org/bukkit/inventory/CartographyInventory.java
+++ b/src/main/java/org/bukkit/inventory/CartographyInventory.java
@@ -3,4 +3,21 @@ package org.bukkit.inventory;
 /**
  * Interface to the inventory of a Cartography table.
  */
-public interface CartographyInventory extends Inventory { }
+public interface CartographyInventory extends Inventory {
+    // Paper begin - add getResult/setResult to CartographyInventory
+    /**
+     * Check what item is in the result slot of this smithing table.
+     *
+     * @return the result item
+     */
+    @org.jetbrains.annotations.Nullable
+    ItemStack getResult();
+
+    /**
+     * Set the item in the result slot of the smithing table
+     *
+     * @param newResult the new result item
+     */
+    void setResult(@org.jetbrains.annotations.Nullable ItemStack newResult);
+    // Paper end - add getResult/setResult to CartographyInventory
+}
